test/test-dev-panel.js

```diff
@@ -316,4 +316,97 @@ exports["test createView panel"] = test(function*(assert) {
   yield closeToolbox();
 });
 
+exports["test destroy tool"] = test(function*(assert) {
+  const MyPanel = Class({
+    extends: Panel,
+    label: "destroy-tool",
+    tooltip: "destroy-tool",
+    icon: iconURI,
+    url: "data:text/html;charset=utf-8,destroy-tool"
+  });
+
+  const myTool = new Tool({
+    panels: {
+      myPanel: MyPanel
+    }
+  });
+
+  const toolbox = yield openToolbox(MyPanel);
+  const myPanel = yield getCurrentPanel(toolbox);
+
+  assert.ok(myPanel instanceof MyPanel,
+            "current panel is instance of MyPanel");
+
+  myTool.destroy();
+
+  const otherPanel = yield getCurrentPanel(toolbox);
+  assert.ok(!(otherPanel instanceof MyPanel),
+            "current panel is not instance of MyPanel");
+
+  yield closeToolbox();
+
+  // At the moment devtools won't throw if requested panel isn't
+  // present and will just open different one, but we still use
+  // try catch since we don't care if it's exception or different
+  // panel and devtools may start throwing in a future.
+  try {
+    const toolbox = yield openToolbox(MyPanel);
+    const otherPanel = yield getCurrentPanel(toolbox);
+
+    assert.ok(!(otherPanel instanceof MyPanel),
+              "requested panel was not open as it was unregistered");
+  } catch (error) {
+    assert.pass("failed to open toolbox with panel that was unregistered");
```

Comment:
we should evaluate the error message here to make sure it is what we expect it to be.